gax-java/gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -98,8 +91,28 @@ public static String getGaxVersion() {
     return GAX_VERSION;
   }
 
-  /** Returns the current runtime version */
-  private static String getRuntimeVersion() {
-    return System.getProperty("java.version");
+  /**
+   * Returns the current runtime version. For GraalVM the values in this method will be fetched at
+   * build time and the values should not differ from runtime (executable)
+   */
+  @VisibleForTesting
+  static String getRuntimeVersion() {
+    String javaRuntimeInformation = System.getProperty("java.version");
+
+    // append the vendor information to the java-version if vendor is present.
+    String vendor = System.getProperty("java.vendor");
+    if (!Strings.isNullOrEmpty(vendor)) {
+      javaRuntimeInformation = String.format("%s__%s", javaRuntimeInformation, vendor);
+    }
+
+    // appends the vendor version information to the java-version if vendor version is present.
+    String vendorVersion = System.getProperty("java.vendor.version");
```

Comment:
Is there a case `java.vendor` does not exist but `java.vendor.version` does not? I feel we should only append it if both exist.

Reply:
> we should only append it if both exist.

Right, that's the spec in the document. @ddixit14 Can you take care of this?

Reply:
The change is updated and pushed.